main/chains/index.js

```diff
@@ -272,7 +272,7 @@ class ChainConnection extends EventEmitter {
       const currentPresets = Object.assign({}, presets.default, presets[this.chainId])
       const target = primary.current === 'custom' ? primary.custom : currentPresets[primary.current]
 
-      if (!this.primary.provider || this.primary.currentPrimaryTarget !== target) {
+      if (target && !this.primary.provider || this.primary.currentPrimaryTarget !== target) {
```

Comment:
can we guard against invalid targets here as well? mainly just empty target as it gets interpreted by eth-provider as no targets and defaults to Frame, so we connect back to ourselves

Reply:
also think this logic isn't right, need parentheses around the 2 "or" clauses to get the desired effect